content/en/blog/_posts/2023-12-20-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,101 @@
+
+#### Continue using in-tree provide code
+
+If you wish to continue using Kubernetes with the in-tree cloud provider code, you will need to modify the command line parameters for `kube-apiserver`, `kube-controller-manager`, and `kubelet` to disable the feature gates for `DisableCloudProviders` and `DisableKubeletCloudCredentialProviders`. To do this, add the following command line flag to the arguments for the previously listed commands:
+
+```
+--feature-gates=DisableCloudProviders=false,DisableKubeletCloudCredentialProviders=false
+```
+
+_Please note that if you have other feature gate modifications on the command line, they will need to include these 2 feature gates._
+
+### Upgrading on other providers
+
+For providers other than Azure, GCE, or vSphere, good news, the external cloud controller manager should be already be in use. The code for AWS and OpenStack providers has been removed from Kubernetes before version 1.27 was released. Other providers beyond the 5 listed in this post were never included in Kubernetes and as such they began their life as external cloud controller managers.
+
+### Upgrading from older Kubernetes versions
+
+If you are upgrading from a Kubernetes release older than 1.26, and you are on AWS, Azure, GCE, OpenStack, or vSphere then you will need to enable the `--cloud-provider=external` flag, and follow the advice for installing and running a cloud controller manager for your provider.
+
+Please read the documentation for [Cloud Controller Manager Administration][ccmadmin] and [Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmha]. See below for links to specific cloud provider implementations.
+
+## Where to find a cloud controller manager?
+
+At its core, this post is about the cloud controllers that were previously included in Kubernetes. As these components move out of the core Kubernetes code and into their own repositories it is important to note a few things.
+
+First, SIG Cloud Provider maintains a reference framework for developers who wish to create cloud controller managers for any provider. See the [cloud-provider repository][cloud-provider] for more information about how these controllers work and how to get started creating your own.
+
+Second, there are many cloud controller managers available for Kubernetes. This post is addressing the providers that have been historically included with Kubernetes but are now in the process of being removed. If you need a cloud controller manager for your provider and do not see it listed here, please reach out to the SIG Cloud Provider community (links below) for help and advice.
```

Comment:
Can we frame it to help readers understand that, in future, the cloud provider integration plugin probably will be open source, but _might not be_?